### ISSUE TITLE
Refactor integration test code

### ISSUE THREAD

**Issue by carlosms** [src-d/engine#347]
Once every PR for #232  is merged we should consider some improvements.

Some examples:
- Tests don't need to be `package cmd`, they don't use any code from `cmd` and should be at least `cmd_test`.
- Since it's integration testing, and not really testing the code in `cmd/srcd/cmd/`, should we move the files to other folder?
- Maybe it's worth to make a common `SetupTest` with the tmpdir creation.
- Improve `IntegrationSuite.CommandContext`, `IntegrationSuite.RunCommand`, `ParseTestSuite.runCommandStdout`...
- Some commands have a wrapper that don't offer any extra functionality, like `IntegrationSuite.RunStop`. For some other commands, we call `RunCommand(context.TODO(), "prune")`. We should either remove the wrappers, or add wrappers for all commands. I prefer to remove them, except when they provide some kind of syntactic sugar or added benefit.
- See if it makes sense to use https://godoc.org/gotest.tools/icmd

**Comment by dpordomingo** [comment]
some nice-to-have, imo

- [ ] https://github.com/src-d/engine/pull/324#discussion_r265705163 separate `stderr` and `stdout`: 
_When running commands in tests, &mdash;like done by [IntegrationSuite::RunCommand](https://github.com/src-d/engine/blob/b72f1fe345a6f035457dfcf4658b92cc65179f45/cmd/test-utils/common.go#L42)&mdash; `stderr` and `stdout` should be separated and assert that nothing is written in `stderr` if the command produced no errors, and the opposite in such cases._
- [ ] https://github.com/src-d/engine/pull/315#discussion_r264764563 more standard way to assert the _Changes done by Engine in its components: 
components outputs could be extracted in a standard way, being able to assert the status of every component in a more readable way (example by https://play.golang.org/p/bC_LNM02Hi-)_
